sdk/samples/iot/paho_iot_provisioning_sas_sample.c

```diff
@@ -351,18 +327,62 @@ static void parse_registration_message(
   IOT_SAMPLE_LOG_SUCCESS("Client received a valid topic response:");
   IOT_SAMPLE_LOG_AZ_SPAN("Topic:", topic_span);
   IOT_SAMPLE_LOG_AZ_SPAN("Payload:", message_span);
-  IOT_SAMPLE_LOG("Status: %d", register_response->status);
+  IOT_SAMPLE_LOG("Status: %d", out_register_response->status);
 
   // Retrieve operation_status.
   if (az_result_failed(
-          rc
-          = az_iot_provisioning_client_parse_operation_status(register_response, operation_status)))
+          rc = az_iot_provisioning_client_parse_operation_status(
+              out_register_response, out_operation_status)))
   {
     IOT_SAMPLE_LOG_ERROR("Failed to parse operation_status: az_result return code 0x%08x.", rc);
     exit(rc);
   }
 }
 
+static void handle_device_registration_status_message(
+    az_iot_provisioning_client_register_response const* register_response,
+    az_iot_provisioning_client_operation_status const* operation_status,
```

Comment:
same here why not just by value

Reply:
Because I wasn't paying attention... lol